File: springboot-02-ioccontainer/src/main/java/com/linkedbear/springboot/assemble/a_module/config/BartenderConfiguration.java
package com.linkedbear.springboot.assemble.a_b_c_module_profile_conditional.config;

import com.linkedbear.springboot.assemble.a_b_c_module_profile_conditional.component.Bartender;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class BartenderConfiguration {
    
    @Bean
    public Bartender zhangxiaosan() {
        return new Bartender("张小三");
    }
    
    @Bean
    public Bartender zhangdasan() {
        return new Bartender("张大三");
    }
}

File: springboot-02-ioccontainer/src/main/kotlin/com/linkedbear/springboot/assemble/a_module/config/BartenderConfiguration.kt
package com.linkedbear.springboot.assemble.a_module.config

import com.linkedbear.springboot.assemble.a_module.component.Bartender
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Profile

@Configuration
@Profile("dev")
open class BartenderConfiguration {
    @Bean
    open fun zhangxiaosan(): Bartender {
        return Bartender("张小三")
    }

    @Bean
    open fun zhangdasan(): Bartender {
        return Bartender("张大三")
    }
}